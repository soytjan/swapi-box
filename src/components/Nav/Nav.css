@import url("https://fonts.googleapis.com/css?family=Bungee+Inline|Roboto|Roboto+Condensed");
nav {
  padding: 20px 0; }
  nav button {
    padding: 5px 12px;
    border-radius: 10px; }

.btn-box {
  width: 400px;
  margin: auto;
  display: flex;
  justify-content: space-between; }

.btn-1 {
  background-color: #6A5671; }

.btn-2 {
  background-color: #41769F; }

.btn-3 {
  background-color: #8DC8AA; }
