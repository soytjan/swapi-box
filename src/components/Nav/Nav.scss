@import "../../variables.scss";

nav {
  padding: 20px 0;
  background-color: black;
  width: 50%;
  border-radius: 25px;
  margin: 35px auto;
  border: 1.5px solid white;

  button {
    padding: 5px 12px;
    border-radius: 10px;
    box-shadow: 0 2px 3px $btn-shadow;

    &:hover {
      cursor: pointer;
    }
  }
}

.btn-box {
  width: 460px;
  margin: auto;
  display: flex;
  justify-content: space-between;
}

.btn-1 {
  background-color: $btn-1-color;

  &:hover {
    background-color: $btn-1-hover;
  }
}

.btn-2 {
  background-color: $btn-2-color;

  &:hover {
    background-color: $btn-2-hover;
  }
}

.btn-3 {
  background-color: $btn-3-color;

  &:hover {
    background-color: $btn-3-hover;
  }
}