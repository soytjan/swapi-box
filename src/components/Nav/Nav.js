import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Nav.css';

const Nav = ({onClick}) => {
  return (
    <nav>
      <div className='btn-box'>
      <button onClick={() => onClick('people')} className='btn-1'><NavLink to='/people'>People</NavLink></button>
      <button onClick={() => onClick('planets')} className='btn-2'><NavLink to='/planets'>Planets</NavLink></button>
      <button onClick={() => onClick('vehicles')} className='btn-3'><NavLink to='/vehicles'>Vehicles</NavLink></button>
      </div>  
    </nav>
  )
}

Nav.propTypes = {
  onClick: PropTypes.func,
};

export default Nav;