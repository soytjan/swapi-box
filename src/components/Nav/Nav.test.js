import React from 'react';
import Nav from './Nav';

describe('Nav', () => {
  it('expect to exist', () => {
    expect(Nav).toBeDefined();
  })
})