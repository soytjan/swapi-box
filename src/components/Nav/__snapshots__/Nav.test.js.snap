// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="btn-box"
>
        <button
                className="btn-1"
                onClick={[Function]}
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/people"
                >
                        People
                </NavLink>
        </button>
        <button
                className="btn-2"
                onClick={[Function]}
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/planets"
                >
                        Planets
                </NavLink>
        </button>
        <button
                className="btn-3"
                onClick={[Function]}
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/vehicles"
                >
                        Vehicles
                </NavLink>
        </button>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn-1"
            onClick={[Function]}
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/people"
            >
                        People
            </NavLink>
</button>,
          <button
            className="btn-2"
            onClick={[Function]}
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/planets"
            >
                        Planets
            </NavLink>
</button>,
          <button
            className="btn-3"
            onClick={[Function]}
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/vehicles"
            >
                        Vehicles
            </NavLink>
</button>,
        ],
        "className": "btn-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/people"
>
              People
</NavLink>,
            "className": "btn-1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "People",
              "to": "/people",
            },
            "ref": null,
            "rendered": "People",
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/planets"
>
              Planets
</NavLink>,
            "className": "btn-2",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Planets",
              "to": "/planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/vehicles"
>
              Vehicles
</NavLink>,
            "className": "btn-3",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Vehicles",
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": [Function],
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="btn-box"
>
          <button
                    className="btn-1"
                    onClick={[Function]}
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/people"
                    >
                              People
                    </NavLink>
          </button>
          <button
                    className="btn-2"
                    onClick={[Function]}
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/planets"
                    >
                              Planets
                    </NavLink>
          </button>
          <button
                    className="btn-3"
                    onClick={[Function]}
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/vehicles"
                    >
                              Vehicles
                    </NavLink>
          </button>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn-1"
              onClick={[Function]}
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/people"
              >
                            People
              </NavLink>
</button>,
            <button
              className="btn-2"
              onClick={[Function]}
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/planets"
              >
                            Planets
              </NavLink>
</button>,
            <button
              className="btn-3"
              onClick={[Function]}
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/vehicles"
              >
                            Vehicles
              </NavLink>
</button>,
          ],
          "className": "btn-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/people"
>
                People
</NavLink>,
              "className": "btn-1",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "People",
                "to": "/people",
              },
              "ref": null,
              "rendered": "People",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/planets"
>
                Planets
</NavLink>,
              "className": "btn-2",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Planets",
                "to": "/planets",
              },
              "ref": null,
              "rendered": "Planets",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/vehicles"
>
                Vehicles
</NavLink>,
              "className": "btn-3",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Vehicles",
                "to": "/vehicles",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
